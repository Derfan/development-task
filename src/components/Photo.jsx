import React from "react";

export default ({location: {state: {photo}}}) => (
    <div>
        <img src={photo.url} alt={photo.title}/>
    </div>
)
