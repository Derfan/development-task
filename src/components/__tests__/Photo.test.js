import React from "react";
import {shallow} from "enzyme";
import Photo from "../Photo";
import {photo} from "./mock";

const props = {
    location: {state: {photo}}
};

describe("Album", () => {
    const component = shallow(<Photo {...props} />);

    it("render", () => {
        expect(component.find("div")).toHaveLength(1);
        expect(component.find("img")).toHaveLength(1);
    });

    it("correct url", () => {
        expect(component.find("img").prop("src")).toEqual(props.location.state.photo.url);
    })
});
