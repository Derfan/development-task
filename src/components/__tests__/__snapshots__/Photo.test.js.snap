// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Album Album snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    location={
      Object {
        "state": Object {
          "photo": Object {
            "albumId": 1,
            "id": 22,
            "thumbnailUrl": "https://via.placeholder.com/150/45601a",
            "title": "et ea illo et sit voluptas animi blanditiis porro",
            "url": "https://via.placeholder.com/600/45601a",
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <img
        alt="et ea illo et sit voluptas animi blanditiis porro"
        src="https://via.placeholder.com/600/45601a"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "alt": "et ea illo et sit voluptas animi blanditiis porro",
        "src": "https://via.placeholder.com/600/45601a",
      },
      "ref": null,
      "rendered": null,
      "type": "img",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <img
          alt="et ea illo et sit voluptas animi blanditiis porro"
          src="https://via.placeholder.com/600/45601a"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "et ea illo et sit voluptas animi blanditiis porro",
          "src": "https://via.placeholder.com/600/45601a",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
